### PR TITLE
please check

### PR DESCRIPTION
**Pull request by jerrycaffe**

### CODE REVIEW

File: src/main/java/com/exercisetracker/exercisetracker/exceptions/CustomExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ControllerAdvice
+public class CustomExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(Exception.class)
```
Comment (kreativecoder) [inline src/main/java/com/exercisetracker/exercisetracker/exceptions/CustomExceptionHandler.java:20]: Do you understand what these means abi na copy and paste :thinking:

File: src/main/java/com/exercisetracker/exercisetracker/model/Exercise.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    private int duration;
+
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+    @JsonFormat(pattern = "dd/MM/yyyy")
+    private LocalDate date;
+
+    @NotNull(message = "Description cannot be empty")
+    private String description;
+
+    private String username;
```
Comment (kreativecoder) [inline src/main/java/com/exercisetracker/exercisetracker/model/Exercise.java]: Why not `id` against `username` that can change?

File: src/main/java/com/exercisetracker/exercisetracker/service/ExerciseService.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+
+    public Exercise saveExercise(Exercise exercise) {
+        return exerciseRepository.save(exercise);
+    }
+
+    public List getExerciseLog(String id) {
+        String username = userService.findUserById(id).getUsername();
+        List exerciseLogs = exerciseRepository.findByUsername(username);
```
Comment (kreativecoder) [inline src/main/java/com/exercisetracker/exercisetracker/service/ExerciseService.java]: If you used `userId`, you'll not need to do double database calls here to get username first. Such calls are expensive!

**Review by kreativecoder — CHANGES_REQUESTED** [review]
Almost there, just a few more changes and we should be good.

And you can start scattering spay codebase...

File: src/main/java/com/exercisetracker/exercisetracker/exceptions/CustomExceptionHandler.java (hunk 1 above)
Comment (jerrycaffe) [reply src/main/java/com/exercisetracker/exercisetracker/exceptions/CustomExceptionHandler.java:20]: Atleast I have a basic knowledge of how it works and I understand spring AOP from the spring tutorial I did

File: src/main/java/com/exercisetracker/exercisetracker/controller/UserController.java [hunk 4]
```diff
@@ -1,14 +1,10 @@
 package com.exercisetracker.exercisetracker.controller;
 
-import com.exercisetracker.exercisetracker.dto.UserRequest;
-import com.exercisetracker.exercisetracker.model.Exercise;
 import com.exercisetracker.exercisetracker.model.User;
-import com.exercisetracker.exercisetracker.repository.UserRepository;
+;
```
Comment (kreativecoder) [inline src/main/java/com/exercisetracker/exercisetracker/controller/UserController.java:4]: Wetin be this one :smirk:

File: src/main/java/com/exercisetracker/exercisetracker/model/Exercise.java [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 import com.fasterxml.jackson.annotation.JsonFormat;
 import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.DBRef;
```
Comment (kreativecoder) [inline src/main/java/com/exercisetracker/exercisetracker/model/Exercise.java:5]: How you take smuggle this one in

File: src/main/java/com/exercisetracker/exercisetracker/repository/ExerciseRepository.java [hunk 6]
```diff
@@ -9,5 +9,7 @@
 
 public interface ExerciseRepository extends MongoRepository<Exercise, String> {
 
-    List<Exercise> findByUsername(String username);
+    List<Exercise> findByUserId(String userId);
+
+//    List findAll(String id);
```
Comment (kreativecoder) [inline src/main/java/com/exercisetracker/exercisetracker/repository/ExerciseRepository.java:14]: Bad practise, remove unused code.

File: src/main/java/com/exercisetracker/exercisetracker/service/ExerciseService.java [hunk 7]
```diff
@@ -18,20 +19,20 @@
     private UserService userService;
 
     public Exercise saveExercise(String id, Exercise exerciseForm) {
-        String username = userService.findUserById(id).getUsername();
+//        String username = userService.findUserById(id).getUsername();
```
Comment (kreativecoder) [inline src/main/java/com/exercisetracker/exercisetracker/service/ExerciseService.java:22]: Bad practise, remove unused code.

**Review by kreativecoder — APPROVED** [review]
Just some ignorable beans...enjoy your new adventure.

I'll still want you to do something with relational db though, specifically Postgres.

**Pull request merged**